droughtpipeline/load.py

```diff
@@ -0,0 +1,872 @@
+            
+
+            if (forecast_climateregion.get_climate_region_data_unit(
+                climate_region_code, lead_time
+                ).alert_class!= "no"):
+                    events[lead_time] = "alert"
+
+            if (forecast_climateregion.get_climate_region_data_unit(
+                climate_region_code, lead_time
+                ).triggered):
+                    events[lead_time] = "trigger"
+                    triggered_lead_times.append(lead_time)
+
+ 
+            if not events:
+                continue
+
+            events = dict(sorted(events.items()))           
+
+
+            # START EVENT LOOP
+            #for climate_region_code in climate_region_codes:#forecast_data.get_climate_region_codes():
+
+            for lead_time_event, event_type in events.items():
+
+                # set as alert if lead time is greater than trigger_on_lead_time
+                if lead_time_event > trigger_on_lead_time and event_type == "trigger":
+                    event_type = "alert"
+                
+
+
+                logging.info(
+                    f"event {event_name}, type '{event_type}', lead time {lead_time_event}"
+                )
+
+                
+
+                pcodes=threshold_climateregion.get_data_unit(climate_region_code).pcodes      
+
+                # send exposure data: admin-area-dynamic-data/exposure
+                indicators = [
+                    "population_affected",
+                    "population_affected_percentage",
+                    #"alert_threshold",
+                    "forecast_trigger",
+                    "forecast_severity",
+                ]
+
+                for indicator in indicators:
+                    for adm_level in admin_levels:
+                        exposure_pcodes = []
+                        for pcode in pcodes[adm_level]: # this admin level should come from config file? 
+                            forecast_admin = forecast_data.get_data_unit(
+                                pcode, lead_time_event
+                            )
+                            amount = None
+                            if indicator == "population_affected":
+                                amount = forecast_admin.pop_affected
+                            elif indicator == "population_affected_percentage":
+                                amount = forecast_admin.pop_affected_perc
+                            elif indicator == "forecast_severity":
+                                amount = (
+                                        1 if event_type == "trigger" else 0
+                                    )
+                            elif indicator == "forecast_trigger":
+                                amount = forecast_trigger_status(
+                                    triggered=(
+                                        True if event_type == "trigger" else False
+                                    ),
```

Comment:
I'm struggling to understand it this correctly implements the instructions as described in https://dev.azure.com/redcrossnl/IBF/_workitems/edit/33291/.

Is event_type a variable that can change between admin-areas? Thereby being true for the warned/triggered areas and false for the others? Or are already only warned/triggered areas included in the data, and is it therefore fine to give all areas either 1 or 0? (This question equally applies to the calculation of 'forecast_severity' right above this one.)

Reply:
The trigger sttus or forecast severity will be the same for all admin areas in a climate region, which is the spatial unit for the drought trigger threshold/model. the calculation for this already done in a previous module, extract.py and forecast.py